shopBuddy/shopBuddy/ItemDetailViewController.m

```diff
@@ -39,9 +39,11 @@ - (void)viewDidLoad {
     }
     __weak __typeof__(self) weakSelf = self;
     [ShoppingList fetchListsByUser:[PFUser currentUser] withCompletion:^(NSArray *lists, NSError *error) {
-        __strong __typeof(weakSelf)strongSelf = weakSelf;
-        strongSelf->lists = lists;
-        [weakSelf _makeMenu];
+        if(weakSelf) {
+            __strong __typeof(weakSelf)strongSelf = weakSelf;
```

Comment:
`weakSelf` has no risk, the issue is with the strong ref, it should be like this instead:
`
        __strong __typeof(weakSelf)strongSelf = weakSelf;
        if(strongSelf) {
        ...
`


`